examples/collector/README.md

```diff
@@ -0,0 +1,144 @@
+```
+
+* The [plugin-name] folder (for example `rand`) will include all files to implement the appropriate interface methods
+* Your [plugin-name] folder  will also include your test files.
+
+
+
+## Interface Methods
+
+In order to write a plugin for Snap, it is necessary to define a few methods to satisfy the appropriate interfaces. These interfaces must be defined for a collector plugin:
+
+
+```go
+// Plugin is an interface shared by all plugins and must implemented by each plugin to communicate with Snap.
+type Plugin interface {
+	GetConfigPolicy() (ConfigPolicy, error)
+}
+
+// Collector is a plugin which is the source of new data in the Snap pipeline.
+type Collector interface {
+	Plugin
+
+	GetMetricTypes(Config) ([]Metric, error)
+	CollectMetrics([]Metric) ([]Metric, error)
+}
+```
+The interface is slightly different depending on what type (collector, processor, or publisher) of plugin is being written. Please see other plugin types for more details.
+
+
+
+## Starting a plugin
+
+After implementing a type that satisfies one of {collector, processor, publisher} interfaces, all that is left to do is to call the appropriate plugin.StartX() with your plugin specific meta options. For example with no meta options specified:
+
+```go
+	plugin.StartCollector(rand.RandCollector{}, pluginName, pluginVersion)
+```
+
+### Meta options
+
+The available options are defined in [plugin/meta.go](https://github.com/intelsdi-x/snap-plugin-lib-go/tree/master/v/1/plugin/meta.go). You can use some or none of the options. The options with definitions/explanations are below:
+
+```go
+// ConcurrencyCount is the max number of concurrent calls the plugin
+// should take.  For example:
+// If there are 5 tasks using the plugin and its concurrency count is 2,
+// snapd will keep 3 plugin instances running.
+// ConcurrencyCount overwrites the default (5) for a Meta's ConcurrencyCount.
+func ConcurrencyCount(cc int) MetaOpt {
+}
+
+// Exclusive == true results in a single instance of the plugin running
+// regardless of the number of tasks using the plugin.
+// Exclusive overwrites the default (false) for a Meta's Exclusive key.
+func Exclusive(e bool) MetaOpt {
+}
+
+// Unsecure results in unencrypted communication with this plugin.
+// Unsecure overwrites the default (false) for a Meta's Unsecure key.
+func Unsecure(e bool) MetaOpt {
+}
+
+// RoutingStrategy will override the routing strategy this plugin requires.
+// The default routing strategy is Least Recently Used.
+// RoutingStrategy overwrites the default (LRU) for a Meta's RoutingStrategy.
+func RoutingStrategy(r router) MetaOpt {
+}
+
+// CacheTTL will override the default cache TTL for the this plugin. snapd
+// caches metrics on the daemon side for a default of 500ms.
+// CacheTTL overwrites the default (500ms) for a Meta's CacheTTL.
+func CacheTTL(t time.Duration) MetaOpt {
+}
+```
+
+An example of what using all of them would look like:
+
+```go
+	plugin.StartCollector(mypackage.Mytype{},
```

Comment:
Those are just placeholders for whatever values a plugin author would want to put there. Maybe it would read more clearly as:

```go

// An example using some arbitrary values:

        plugin.StartCollector(
                mypackage.Mytype{},
                pluginName,
 				pluginVersion,
 				plugin.ConcurrencyCount(2),
 				plugin.Exclusive(true),
 				plugin.Unsecure(true),
 				plugin.RoutingStrategy(StickyRouter),
 				plugin.CacheTTL(time.Second))
```

Which both gives example of legit values and calls out that they are arbitrarily selected.